device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/MultiplexClientManager.java

```diff
@@ -0,0 +1,54 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeCallback;
+import com.microsoft.azure.sdk.iot.device.MultiplexingClient;
+import lombok.experimental.Delegate;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+
+/**
+ * This class is in charge of handling reconnection logic and registering callbacks for connection status changes.
+ * It will delegate all other calls other than `Open`, `Close` and registerConnectionStatusChangeCallbaack to the inner client (MultiplexingClient)
+ */
+@Slf4j
+public class MultiplexClientManager extends ClientManagerBase {
+
+    // Define method calls that will not be delegated to the inner client.
+    private interface DeviceClientNonDelegatedFunction {
+        void open();
+        void close();
+        void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext);
+    }
+
+    // The methods defined in the interface DeviceClientNonDelegatedFunction will be called on MultiplexingClientManager, and not on MultiplexingClient.
+    @Delegate(excludes = DeviceClientNonDelegatedFunction.class)
+    private final MultiplexingClient client;
+    private final String multiplexClientId;
+
+    MultiplexClientManager(MultiplexingClient multiplexingClient, String multiplexClientId) {
```

Comment:
Add a header here if only to clarify that the multiplexClientId is user-defined, and isn't some Id like deviceId where they get it from the portal.